autoload/maktaba/selector.vim

```diff
@@ -0,0 +1,428 @@
+  return self
+endfunction
+
+
+""
+" @dict Selector.WithSyntax
+" Configures an {ApplySyntax} function to be called in the selector window.
+" This will by applied in addition to standard syntax rules for rendering the
+" help header, etc.
+function! maktaba#selector#DoWithSyntax(ApplySyntax) dict abort
+  let self._ApplySyntax = maktaba#ensure#IsCallable(a:ApplySyntax)
+  return self
+endfunction
+
+
+""
+" @dict Selector.WithExtraOptions
+" Configures {ApplyExtraOptions} for additional window-local settings for
+" selector window.
+" If not configured, the default extra options just disable 'number'.
+function! maktaba#selector#DoWithExtraOptions(ApplyExtraOptions) dict abort
+  let self._ApplyExtraOptions = maktaba#ensure#IsCallable(a:ApplyExtraOptions)
+  return self
+endfunction
+
+
+""
+" @dict Selector.WithName
+" Configures {name} to show as the window name on the selector.
+" If not configured, the default name is "__SelectorWindow__".
+function! maktaba#selector#DoWithName(name) dict abort
+  let self._name = maktaba#ensure#IsString(a:name)
+  return self
+endfunction
+
+
+""
+" @dict Selector.Show
+" Shows a selector window for the @dict(Selector) with [minheight], [maxheight],
+" and [position].
+" @default minheight=5
+" @default maxheight=25
+" @default position='botright'
+function! maktaba#selector#DoShow(...) dict abort
+  let l:min_win_height = (a:0 >= 1 && a:1 isnot -1) ?
+      \ maktaba#ensure#IsNumber(a:1) : 5
+  let l:max_win_height = (a:0 >= 2 && a:2 isnot -1) ?
+      \ maktaba#ensure#IsNumber(a:2) : 25
+  let l:position = maktaba#ensure#IsString(get(a:, 3, 'botright'))
+
+  " Show one empty line at the bottom of the window.
+  " (2 is correct -- I know it looks bizarre)
+  let l:win_size = len(self._infolist) + 2
+  if l:win_size > l:max_win_height
+    let l:win_size = l:max_win_height
+  elseif l:win_size < l:min_win_height
+    let l:win_size = l:min_win_height
+  endif
+
+  let s:current_savedview = winsaveview()
+  let s:curpos_holder = getpos(".")
+  let s:last_winnum = winnr()
+
+  " Open the window in the specified window position.  Typically, this opens up
+  " a flat window on the bottom (as with split).
+  execute l:position l:win_size 'new'
+  let s:selectors_by_buffer_number[bufnr('%')] = self
+  call s:SetWindowOptions(self)
+  silent execute 'file' self._name
+  let [l:lines, l:data] = s:SplitLinesAndData(self._infolist)
+  let b:selector_lines_data = l:data
+  call s:InstantiateKeyMaps(self._mappings)
+  setlocal noreadonly
+  setlocal modifiable
+  call maktaba#buffer#Overwrite(1, line('$'), l:lines)
+  " Add the help comments at the top (do this last so cursor stays below it).
+  call append(0, self._GetHelpLines())
+  setlocal readonly
+  setlocal nomodifiable
+
+  " Restore the previous windows view
+  let l:buffer_window = winnr()
+  call maktaba#selector#ReturnToWindow()
+  call winrestview(s:current_savedview)
+  execute l:buffer_window  'wincmd w'
+
+  return self
+endfunction
+
+
```

Comment:
The spacing between functions is really inconsistent. Sometimes it's one blank line and sometimes it's two.

Reply:
Updated to use 2 everywhere like https://google.github.io/styleguide/vimscriptfull.xml?showone=Functions#Functions recommends.